src/kernel/include/shared/cos_types.h

```diff
@@ -235,16 +236,17 @@ static inline unsigned long captbl_idsize(cap_t c)
  * 2GB-> = system physical memory
  */
 enum {
-	BOOT_CAPTBL_SRET       = 0,
-	BOOT_CAPTBL_SELF_CT    = 4,
-	BOOT_CAPTBL_SELF_PT    = 6,
-	BOOT_CAPTBL_SELF_COMP  = 8,
-	BOOT_CAPTBL_BOOTVM_PTE = 12,
-	BOOT_CAPTBL_PHYSM_PTE  = 14,
-	BOOT_CAPTBL_KM_PTE     = 16,
-
-	BOOT_CAPTBL_COMP0_CT           = 18,
-	BOOT_CAPTBL_COMP0_PT           = 20,
+	BOOT_CAPTBL_SRET          = 0,
+	BOOT_CAPTBL_SELF_CT       = 4,
+	BOOT_CAPTBL_SELF_PT       = 6,
+	BOOT_CAPTBL_SELF_COMP     = 8,
+	BOOT_CAPTBL_BOOTVM_PTE    = 12,
+	BOOT_CAPTBL_SELF_LOCAL_PT = 14,
```

Comment:
"Isn't `pgtbl` of a component supposed to have all the pages that it has access to?"
Here "access" is a little ambiguous.
Take memory manager for example, it manages 1G memory. This does not mean the manager will actually use that 1G memory. Broadly there are two kinds of memory:
memory is used by the manager for its code data. Those memory are present in the page able referenced by SELF_PT.
other memory that is managed by the manager, but not actually accessed by the manager.
those memory are present in SELF_UNTYPED_PT.

"user can call the kernel api with different comp_info appropriately?"
This is also a little ambiguous.
A client component invokes a manager component without providing comp_info.
The manager finds the comp_info of that client based on its component id or other similar identifier.
It is possible for manager to maintain separate comp_info for each client. But I think comp_info is heavy weight in the case, as only we care about memory resource. comp_info also includes information about cap tables.